File: client-java/controller/src/main/java/org/evomaster/client/java/controller/mongo/operations/NotInOperation.java
package org.evomaster.client.java.controller.mongo.operations;

import java.util.List;

/**
 * Represent $nin operation.
 * Selects the documents where:
 * - the field value is not in the specified array
 * - the field does not exist.
 */
public class NotInOperation<V> extends QueryOperationWithField {
    private final List<V> values;

    public NotInOperation(String fieldName, List<V> values) {
        super(fieldName);
        this.values = values;
    }

    public List<V> getValues() {
        return values;
    }
}

File: client-java/controller/src/main/java/org/evomaster/client/java/controller/mongo/operations/NotInOperation.kt
package org.evomaster.client.java.controller.mongo.operations

/**
 * Represent $nin operation.
 * Selects the documents where:
 *  - the field value is not in the specified array
 *  - the field does not exist.
 */
open class NotInOperation<V>(open val fieldName: String, open val values: ArrayList<V>) : QueryOperation()